### PR TITLE
Bug 1813775: Publish license file

### PR DESCRIPTION
**Pull request by jhadvig**
Based on:
- https://github.com/openshift/console-operator/pull/386
- https://github.com/openshift/console-operator/pull/412

Addressing comments + rebasing on top of master and fixing tests, so we can start backport process.

Screen:
Screenshot 2020-06-15 at 17 22 06

Not sure if it makes sense to backport all four commits, since my last on removes most of the added code. Would make the backport a lot easier.

/assign @spadgett

### CODE REVIEW

File: pkg/console/controllers/clidownloads/controller.go [hunk 1]
```diff
@@ -186,6 +186,11 @@ func PlatformBasedOCConsoleCLIDownloads(host, cliDownloadsName string) *v1.Conso
 		})
 	}
 
+	links = append(links, v1.CLIDownloadLink{
+		Href: fmt.Sprintf("%s/oc-license", baseURL),
+		Text: fmt.Sprintf("Download LICENSE for OpenShift Command Line Interface"),
```
Comment (spadgett) [inline pkg/console/controllers/clidownloads/controller.go:191]: Maybe just "LICENSE"? Shouldn't need `Sprintf`

```suggestion
		Text: "LICENSE",
```

File: pkg/console/controllers/clidownloads/controller_test.go [hunk 2]
```diff
@@ -135,6 +135,10 @@ The oc binary offers the same capabilities as the kubectl binary, but it is furt
 							Href: "https://www.example.com/s390x/linux/oc.tar",
 							Text: "Download oc for Linux for IBM Z (unsupported)",
 						},
+						{
+							Href: "https://www.example.com/oc-license",
+							Text: "Download LICENSE for OpenShift Command Line Interface",
```
Comment (spadgett) [inline pkg/console/controllers/clidownloads/controller_test.go:140]:
```suggestion
							Text: "LICENSE",
```

**Comment by jhadvig** [comment]
@spadgett comments addressed.  Just FYI there is a `TestDownloadsEndpoint` that is testing the license file endpoint.

**Pull request merged**